frontend/ui/elements/screen_android.lua

```diff
@@ -32,13 +32,35 @@ local function humanReadableTimeout(timeout)
     end
 end
 
+local function canModifyTimeout(timeout)
+    if needs_wakelocks then return false end
+    if timeout == system or timeout == screenOn then
+        return true
+    else
+        return android.settings.canWrite()
+    end
+end
+
 local function timeoutEquals(timeout)
-    return timeout == android.getScreenOffTimeout()
+    return timeout == android.timeout.get()
 end
 
 local function saveAndApplyTimeout(timeout)
     G_reader_settings:saveSetting("android_screen_timeout", timeout)
-    android.setScreenOffTimeout(timeout)
+    android.timeout.set(timeout)
+end
+
+local function requestWriteSettings()
+    local UIManager = require("ui/uimanager")
+    local ConfirmBox = require("ui/widget/confirmbox")
+    UIManager:show(ConfirmBox:new{
+        text = _("Allow KOReader to modify system settings?\n\nYou will be prompted with a permission management screen. You'll need to give permission to KOReader and start the program again"),
```

Comment:
Maybe something like this? (Btw, you missed the period at the end. ;-) )
```suggestion
        text = _("Allow KOReader to modify system settings?\n\nYou will be prompted with a permission management screen. You'll need to give KOReader permission and then restart the program."),
```

Reply:
thanks!